test/tests/ruby-bundler/container.sh

```diff
@@ -12,3 +12,14 @@ BUNDLE_FROZEN=0 bundle install
 cp Gemfile.lock Gemfile.lock.orig
 BUNDLE_FROZEN=1 bundle install
 diff -u Gemfile.lock.orig Gemfile.lock >&2
+
+new_bundler=$(ruby -rbundler -e 'print Gem::Version.new(Bundler::VERSION) >= Gem::Version.new("2.1.a")')
+
+if [ "$new_bundler" = "true" ]
+then
+  BUNDLE_DEPLOYMENT=1 bundle install
+else
+  BUNDLE_DEPLOYMENT=1 bundle install --deployment
+fi
+
+diff -u Gemfile.lock.orig Gemfile.lock >&2
```

Comment:
There's a few things that seem weird here -- why are we using both `BUNDLE_DEPLOYMENT=1` and `--deployment`?  Shouldn't this be using either one or the other?

In other words, something like this:

```bash
if ruby -rbundler -e 'exit Gem::Version.new(Bundler::VERSION) >= Gem::Version.new("2.1") ? 0 : 1'; then
	BUNDLE_DEPLOYMENT=1 bundle install
else
	bundle install --deployment
fi
diff -u Gemfile.lock.orig Gemfile.lock >&2
```

Also, shouldn't this be verifying more after it does `--deployment` to verify that it fully worked properly (more than just verifying that it enabled "frozen"), or is this enough to fail in the cases we need it to?

Reply:
Totally right about using either one or the other, my bad! I'll update the test with your suggestion :purple_heart:

Regarding what to test here, the issue I'm trying to cover is docker-library/ruby#288, where `bundle install --deployment` would abort due to some incompatible settings, so yeah checking for frozen-ness should be enough I think. `--deployment` is essentially an alias to `--frozen --path vendor/bundler`, but the most important part is `--frozen`. There's even plans in bundler to remove the `vendor/bundler` part and just make it an alias to `--frozen`.